packages/generate/test/select.test.ts

```diff
@@ -1,761 +1,731 @@
-    }))
-    .run(client);
+    assert.deepEqual(result?.id, data.spidey.id);
+  });
 
-  expect(result?.id).toEqual(data.spidey.id);
-});
+  test("filter by id expr", async () => {
+    const result = await e
+      .select(e.Hero, () => ({
+        filter_single: { id: e.uuid(data.spidey.id) },
+      }))
+      .run(client);
 
-test("filter by id expr", async () => {
-  const result = await e
-    .select(e.Hero, (hero) => ({
-      filter_single: { id: e.uuid(data.spidey.id) },
-    }))
-    .run(client);
+    assert.deepEqual(result?.id, data.spidey.id);
+  });
 
-  expect(result?.id).toEqual(data.spidey.id);
-});
+  test("limit 1", async () => {
+    const query = e
+      .select(e.Hero, (hero) => ({
+        order_by: hero.name,
+        offset: 1,
+        limit: 1,
+      }))
+      .assert_single();
+    const result = await e.select(query).run(client);
+    assert.deepEqual(result?.id, data.iron_man.id);
+  });
 
-test("limit 1", async () => {
-  const query = e
-    .select(e.Hero, (hero) => ({
+  test("limit 2", async () => {
+    const query = e.select(e.Hero, (hero) => ({
       order_by: hero.name,
       offset: 1,
-      limit: 1,
-    }))
-    .assert_single();
-  const result = await e.select(query).run(client);
-  expect(result?.id).toEqual(data.iron_man.id);
-});
+      limit: 2,
+    }));
+    const results = await query.run(client);
+
+    assert.equal(results.length, 2);
+    assert.deepEqual(results, [
+      { id: data.iron_man.id },
+      { id: data.spidey.id },
+    ]);
+  });
 
-test("limit 2", async () => {
-  const query = e.select(e.Hero, (hero) => ({
-    order_by: hero.name,
-    offset: 1,
-    limit: 2,
-  }));
-  const results = await query.run(client);
+  test("order by self", async () => {
+    const query = e.select(e.Hero, (hero) => ({
+      order_by: hero,
+    }));
+    const result = await query.run(client);
+    assert.deepEqual(
+      result,
+      [data.cap, data.spidey, data.iron_man]
+        .map((h) => ({ id: h.id }))
+        .sort((a, b) => a.id.localeCompare(b.id))
+    );
+  });
 
-  expect(results.length).toEqual(2);
-  expect(results).toEqual([{ id: data.iron_man.id }, { id: data.spidey.id }]);
-});
+  test("shapes", async () => {
+    const query = e.select(
+      e
+        .select(e.Hero, (hero) => ({
+          filter: e.op(hero.name, "=", "Iron Man"),
+        }))
+        .assert_single(),
+      () => ({
+        id: true,
+        name: true,
+        secret_identity: true,
+        villains: { id: true },
+      })
+    );
 
-test("order by self", async () => {
-  const query = e.select(e.Hero, (hero) => ({
-    order_by: hero,
-  }));
-  const result = await query.run(client);
-  expect(result).toEqual(
-    [data.cap, data.spidey, data.iron_man]
-      .map((h) => ({ id: h.id }))
-      .sort((a, b) => a.id.localeCompare(b.id))
-  );
-});
+    const result = await query.run(client);
+    assert.ok(result);
+    assert.deepEqual(result, { ...result, ...data.iron_man });
```

Comment:
Here's my solution to the loosy `toMatchObject`: merge the expectation with the result.